locations/spiders/raleys_us.py

```diff
@@ -1,40 +1,42 @@
-from scrapy.http import JsonRequest
+from scrapy.http import JsonRequest, Request
 
-from locations.categories import Categories
-from locations.hours import OpeningHours
+from locations.categories import Categories, apply_category
 from locations.json_blob_spider import JSONBlobSpider
 
-BRAND_WIKIDATA = {
-    "Bel Air": "Q112922067",
-    "Nob Hill Foods": "Q121816894",
-    "Raley's": "Q7286970",
-    "Raley's ONE Market": "Q7286970",
+BRAND_MAP = {
+    "Bel Air": {"brand": "Bel Air", "brand_wikidata": "Q112922067"},
+    "Nob Hill Foods": {"name": "Nob Hill Foods", "brand": "Nob Hill Foods"},
+    "Raley's": {"brand": "Raley's", "brand_wikidata": "Q7286970"},
+    "Raley's ONE Market": {"name": "Raley's O-N-E Market", "brand": "Raley's", "brand_wikidata": "Q7286970"},
 }
 
 
 class RaleysUSSpider(JSONBlobSpider):
     name = "raleys_us"
-    item_attributes = {"extras": Categories.SHOP_SUPERMARKET.value}
     custom_settings = {"DOWNLOAD_TIMEOUT": 55, "ROBOTSTXT_OBEY": False}
     locations_key = ["data"]
     allowed_domains = ["www.raleys.com"]
 
     def start_requests(self):
         for domain in self.allowed_domains:
-            yield JsonRequest(
-                f"https://{domain}/api/store", data={"rows": 1000, "searchParameter": {"shippingMethod": "pickup"}}
-            )
+            yield Request(f"https://{domain}/stores", callback=self.start_api_request)
+
+    def start_api_request(self, response):
+        yield JsonRequest(
+            response.urljoin("/api/store"), data={"rows": 1000, "searchParameter": {"shippingMethod": "pickup"}}
+        )
 
     def post_process_item(self, item, response, location):
         item["ref"] = location["number"]
         item["website"] = response.urljoin(f"/store/{location['number']}")
-        item["brand"] = item["name"] = location["brand"]["name"]
-        item["brand_wikidata"] = BRAND_WIKIDATA.get(location["brand"]["name"])
+        if brand := BRAND_MAP.get(location["brand"]["name"]):
+            del item["name"]
+            item.update(brand)
+        elif self.name == "raleys_us":
+            # Only log an error for this spider. FoodCityArizonaUSSpider is a subclass that doesn't use the brand mapping.
+            self.logger.error("Unexpected brand: {}".format(location["brand"]["name"]))
```

Comment:
This is pretty smelly. If we're starting to perform checks like this we probably don't want the subclass relationship.

Reply:
I didn't know it was using this class when I added it, and I agree it's not a good idea. Whenever we are doing anything specific in here, it shouldn't be used by multiple spiders, however since we are already applying a category, I've embraced it and added the Food City branding here instead of the other spider.